tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3767 @@
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 2</strong> to denote a <i>Metadata Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data Length</td>
+                                <td>uint16</td>
+                                <td>
+                                    Length of the following data field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data</td>
+                                <td>ByteArray</td>
+                                <td>Binary data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                    <details>
+                        <summary>Tag Block</summary>
+                        <blockquote>
+                            Defines an indexation tag to which the output can be indexed by additional node plugins.
+                        </blockquote>
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 3</strong> to denote a <i>Tag Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Tag Length</td>
+                                <td>uint8</td>
+                                <td>
+                                    Length of the following tag field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Tag</td>
+                                <td>ByteArray</td>
+                                <td>Binary indexation data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                </td>
+            </tr>
+        </table>
+    </details>
+</table>
+
+### Additional Transaction Syntactic Validation Rules
+
+- `Amount` field must fulfill the dust protection requirements and must not be `0`.
```

Comment:
To clear out any ambiguity: do we want a maximum for the amount ? I know that as part of the transaction validation, the sum of all outputs must not exceed the total supply, but what about outputs on their own ? Is an output with an amount that exceeds the total supply valid ?

Reply:
Of course such an output makes no sense because the transaction validation will fail immeditely. Therefore, this rule is implicitly there, however not explicitly.

Checking if the amount is greater than max supply is a remnant from the bundle days, where you could have negative `Amount` in transactions, so the bundle could have been balanced even if one transaction had greater `Amount` than max supply.

We could add this extra syntactic check to outputs, but as I wrote above, it is redundant.

Reply:
I've added it to the TIP.